### DISCUSSION TITLE
Data Loss on Container Restart: PostgreSQL Docker Image in TrueNAS Scale

### DISCUSSION THREAD

**Discussion by DenisTimofijuk** [docker-library/postgres#1203] [General]
### Issue Description:

I'm encountering an issue with the PostgreSQL Docker Official Image on my TrueNAS Scale setup. While I can install the image without any problems and make changes via pgAdmin, upon restarting the image, all modifications are lost, reverting everything back to its initial state.

### Expected Behavior:

I expect the changes made within the PostgreSQL container to persist across restarts, ensuring data integrity and consistency.

### Current Behavior:

However, it seems that the data is only stored in memory and not being saved to the disks, resulting in loss of changes upon restart.

### Steps to Reproduce:

1. Install the PostgreSQL Docker Official Image on TrueNAS Scale.
2. Connect to the container via pgAdmin.
3. Make changes or create new data within the container.
4. Restart the container.
5. Observe that all changes are lost, reverting to the initial state.

### Additional Information:

- Docker Image Version: latest
- TrueNAS Scale Version: TrueNAS-SCALE-23.10.1.3
- Docker Configuration Settings: default

**Comment by LaurentGoderre** [comment]
Did you create a volume to persist the data? Without a volume the data would be lost when restarting.
